packager/packager.cc

```diff
@@ -0,0 +1,665 @@
+            return tree;
+        }
+
+        // Blacklisted methods
+        if (send->fun == core::Names::extend() || send->fun == core::Names::include()) {
+            if (auto e = ctx.beginError(send->loc, core::errors::Packager::InvalidPackageExpression)) {
+                e.setHeader("Invalid expression in package: `{}` is not allowed", send->fun.data(ctx)->shortName(ctx));
+            }
+            return tree;
+        }
+
+        // Sanity check arguments for unrecognized methods
+        if (send->fun != core::Names::export_() && send->fun != core::Names::import() &&
+            send->fun != core::Names::exportMethods()) {
+            for (const auto &arg : send->args) {
+                if (!ast::isa_tree<ast::Literal>(arg)) {
+                    if (auto e = ctx.beginError(arg->loc, core::errors::Packager::InvalidPackageExpression)) {
+                        e.setHeader("Invalid expression in package: Arguments to functions must be literals");
+                    }
+                }
+            }
+        }
+
+        if (info == nullptr) {
+            // We haven't yet entered the package class.
+            return tree;
+        }
+
+        if (send->fun == core::Names::export_() && send->args.size() == 1) {
+            auto target = verifyConstant(ctx, core::Names::export_(), send->args[0]);
+            // null indicates an invalid export.
+            if (target != nullptr) {
+                exported.emplace_back(copyConstantLit(target));
+                // Transform the constant lit to refer to the target within the mangled package namespace.
+                send->args[0] = prependInternalPackageNameToScope(move(send->args[0]));
+            }
+        }
+
+        if (send->fun == core::Names::import() && send->args.size() == 1) {
+            auto target = verifyConstant(ctx, core::Names::import(), send->args[0]);
+            // null indicates an invalid import.
+            if (target != nullptr) {
+                auto name = getPackageName(ctx, send->args[0]);
+                if (name.mangledName.exists()) {
+                    if (name.mangledName == info->name.mangledName) {
+                        if (auto e = ctx.beginError(target->loc, core::errors::Packager::NoSelfImport)) {
+                            e.setHeader("Package `{}` cannot import itself",
+                                        absl::StrJoin(info->name.fullName, "::", NameFormatter(ctx)));
+                        }
+                    }
+                    info->importedPackageNames.emplace_back(move(name));
+                }
+            }
+        }
+
+        if (send->fun == core::Names::exportMethods()) {
+            const bool alreadyCalled = exportMethodsLoc.exists();
+            if (alreadyCalled) {
+                if (auto e = ctx.beginError(send->loc, core::errors::Packager::MultipleExportMethodsCalls)) {
+                    e.setHeader("export_methods can only be called once in a package");
```

Comment:
```suggestion
                    e.setHeader("`{}` can only be called once in a package", "export_methods");
```

This will add colors.

Reply:
ah, TIL